File: sol/2799/2799.py
class Solution:
    def countCompleteSubarrays(self, nums: List[int]) -> int:
        n = len(nums)
        totalUniq = len(frozenset(nums))
        freq = [0] * 2001
        l = 0
        res = 0
        uqCount = 0
        for r in range(n):
            freq[nums[r]] += 1
            if freq[nums[r]] == 1:
                uqCount += 1
            while l <= r and uqCount == totalUniq:
                freq[nums[l]] -= 1
                if freq[nums[l]] == 0:
                    uqCount -= 1
                l += 1
            res += l
        return res

File: sol/2799/2799.cpp
#include <unordered_set>

struct Solution {
    int countCompleteSubarrays(vector<int>& nums) {
        int n = nums.size(), l = 0, res = 0, uqCount = 0;
        int freq[2001] = {0};
        unordered_set<int> uniqSet;
        uniqSet.insert(nums.begin(), nums.end());
        int totalUniq = uniqSet.size();
        for (int r = 0; r < n; ++r){
            if (++freq[nums[r]] == 1){
                ++uqCount;
            }
            while (l <= r && uqCount == totalUniq){
                if (--freq[nums[l++]] == 0){
                    --uqCount;
                }
            }
            res += l;
        }
        return res;
    }
};
